### PR TITLE
Add hook LIST.make

### LINKED ISSUE #2253: Missing hook: LIST.make
With this small k definition:
```
requires "domains.md"

module TEST
    imports DOMAINS
    imports ARRAY

    configuration
        <k> $PGM:KItem </k>

endmodule
```

Trying to krun `arr ( .List , 10 , 0 ) [0]` returns `#Bottom`. The LLVM backend will return 0.

Trying to krun `arr ( .List , 10 , 0 ) [0 <- 1]` gives an error for a missing LIST.make hook.

It looks like it's these rules in `domains.md:164` for the array update:
```
  syntax List ::= ensureOffsetList(List, Int, KItem) [function]
  rule ensureOffsetList(L::List, IDX::Int, D::KItem) => #if IDX >=Int size(L) #then updateList(makeList(IDX +Int 1, D), 0, L) #else L #fi

  rule arr(L::List, I::Int, D::KItem) [ IDX::Int <- VAL::KItem ] => arr(ensureOffsetList(L, IDX, D) [ IDX <- VAL ], I, D)
```

### PR DESCRIPTION
**Pull request by andreiburdusa**
---
Fixes #2253 

~~Needs answer for: https://runtimeverification.slack.com/archives/CC360GUTG/p1605864619247900?thread_ts=1605699058.240400&cid=CC360GUTG~~

###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`

### CODE REVIEW

File: kore/src/Kore/Builtin/List.hs [hunk 1]
```diff
@@ -324,6 +327,15 @@ evalSize resultSort [_list] = do
         & return
 evalSize _ _ = Builtin.wrongArity sizeKey
 
+evalMake :: Builtin.Function
+evalMake resultSort [_len, value] = do
+    _len <- fromInteger <$> Int.expectBuiltinInt getKey _len
+    let len
+          | _len < 0 = 0
```
Comment (ttuegel) [inline kore/src/Kore/Builtin/List.hs:334]: Is there a test of this behavior anywhere, like the frontend integration tests?

Comment (andreiburdusa) [reply kore/src/Kore/Builtin/List.hs:334]: > Is there a test of this behavior anywhere, like the frontend integration tests?

I don't think so. Should that be an error instead?

**Comment by andreiburdusa** [comment]
I merged the branch of [LIST.updateAll](https://github.com/kframework/kore/pull/2275) and now the test from the issue works like on the llvm backend.

File: kore/src/Kore/Builtin/List.hs [hunk 2]
```diff
@@ -324,6 +330,38 @@ evalSize resultSort [_list] = do
...
+    _list2 <- expectBuiltinList getKey _list2
+    _ix <- fromInteger <$> Int.expectBuiltinInt getKey _ix
+    let len1 = Seq.length _list1
+        len2 = Seq.length _list2
+        ix
+            | _ix < 0 =
+            -- negative indices count from end of list
+            _ix + len1
+            | otherwise = _ix
+    if ix >= 0 && ix < len1 && ix + len2 <= len1
```
Comment (ttuegel) [inline kore/src/Kore/Builtin/List.hs:354]: What does the other backend do if `ix = len1`, but `len2 = 0`?

File: kore/test/Test/Kore/Builtin/List.hs [hunk 3]
```diff
@@ -417,6 +419,91 @@ test_size =
         (===) Pattern.top    =<< evaluateT predicate
     ]
 
+test_make :: [TestTree]
```
Comment (ttuegel) [inline kore/test/Test/Kore/Builtin/List.hs:422]: I don't think the new property tests are useful compared to having a list of simple test cases.

**Review by ttuegel — CHANGES_REQUESTED** [review]

**Review by ttuegel — APPROVED** [review]

**Pull request merged**